packages/react-data-grid/src/utils/columnUtils.ts

```diff
@@ -64,22 +65,45 @@ export function getColumnMetrics<R>(metrics: Metrics<R>): ColumnMetrics<R> {
   };
 }
 
+// get column width, fallback on defaultWidth when width is undefined
+function getWidth(
+  viewportWidth: number,
+  minWidth?: number | string,
+  defaultWidth?: number
+): number | void {
+  switch (typeof minWidth) {
+    case 'string':
+      return /^\d+%$/.test(minWidth)
+        ? Math.floor((viewportWidth * parseInt(minWidth, 10)) / 100)
+        : defaultWidth;
+    case 'number':
+      return minWidth;
+    default:
+      return defaultWidth;
+  }
+}
+
 function getSpecifiedWidth<R>(
   column: Column<R>,
   columnWidths: Map<keyof R, number>,
   viewportWidth: number,
-  minColumnWidth: number
+  defaultMinColumnWidth: number
 ): number | void {
+  // get column min width from grid or column prop, fallback on defaultMinColumnWidth if undefined
+  const minWidth = typeof column.minWidth === 'number' ? column.minWidth : defaultMinColumnWidth;
+  const width = getWidth(viewportWidth, column.width);
+
+  // SelectColumn must always use width prop when defined
+  if (column.key === SelectColumn.key) {
+    return width || minWidth;
+  }
```

Comment:
We shouldn't treat `SelectColumn` as a special case.

We only need to update this function that way:
```ts
  if (typeof column.width === 'number') {
    return column.width;
  }
  if (typeof column.width === 'string' && /^\d+%$/.test(column.width)) {
    return Math.max(Math.floor(viewportWidth * parseInt(column.width, 10) / 100), column.minWidth || minColumnWidth);
  }
  if (typeof column.minWidth === 'number' && column.minWidth > minColumnWidth) {
    return column.minWidth;
  }
```

The new `if` at the end should allow us to properly handle `unallocatedWidth` in `getColumnMetrics`.
For example:
- you have 3 columns with undefined widths, one column has a `minWidth` of `300`
- the container is 500px wide
- `minColumnWidth` is set to `80`
- you'll get 1 column with `300px` width, and the 2 other columns will be `100px` wide.